community/index/src/main/java/org/neo4j/index/gbptree/SeekCursor.java

```diff
@@ -362,8 +384,17 @@ public boolean next() throws IOException
                     bTreeNode.valueAt( cursor, mutableValue, pos );
                 }
             }
-            while ( rediscoverKeyPosition = cursor.shouldRetry() );
+            while ( needToRediscoverKeyPosition = cursor.shouldRetry() );
             checkOutOfBounds( cursor );
+            if ( expectingShouldRetry )
+            {
+                throw new TreeInconsistencyException( "Read inconsistent tree node %d%n" +
```

Comment:
Hmm, actually no, the two places have different data in their exceptions. I don't think they can be merged.